src/main/java/com/laytonsmith/core/functions/Echoes.java

```diff
@@ -401,6 +401,11 @@ public Mixed exec(Target t, Environment env, Mixed... args) throws ConfigRuntime
 				//If the value is empty string, set the color to white.
 				color = MCChatColor.WHITE.toString();
 			}
+
+			if(a.matches("(?i)^#[0-9A-F]{6}$")) {
```

Comment:
It's actually slightly different, the one in `fromRGBValue` is checking without the `#` as it's optional. Could maybe become a `.startsWith("#")` though here instead of checking the regex.

The one here and the one in `colorize()` are the same though, but I couldn't see an easy way to reduce that duplication.

Reply:
What I mean is something like:
```java
String rgbColor = MCChatColor.fromRGBValue(a);
if(rgbColor != null) {
    color = rgbColor;
}
```
As long as `fromRGBValue` returns `null` for invalid RGB/hex colors, that should work.

Reply:
Oh, fair enough. I've pushed an update that reduces all three into the one, and makes the `#` non optional. Makes it simpler, thanks.